src/app/MessageDef.cpp

```diff
@@ -0,0 +1,3549 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2018 Google LLC.
+ *    Copyright (c) 2016-2017 Nest Labs, Inc.
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+/**
+ *    @file
+ *      This file defines CHIP interaction model message parsers and encoders
+ *      Interaction model profile.
+ *
+ */
+
+// __STDC_FORMAT_MACROS must be defined for PRIX64 to be defined for pre-C++11 clib
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif // __STDC_FORMAT_MACROS
+
+// __STDC_LIMIT_MACROS must be defined for UINT8_MAX and INT32_MAX to be defined for pre-C++11 clib
+#ifndef __STDC_LIMIT_MACROS
+#define __STDC_LIMIT_MACROS
+#endif // __STDC_LIMIT_MACROS
+
+// __STDC_CONSTANT_MACROS must be defined for INT64_C and UINT64_C to be defined for pre-C++11 clib
+#ifndef __STDC_CONSTANT_MACROS
+#define __STDC_CONSTANT_MACROS
+#endif // __STDC_CONSTANT_MACROS
+
+#include "MessageDef.h"
+#include <algorithm>
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdio.h>
+
+using namespace chip;
+using namespace chip::TLV;
+
+#ifndef CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK
+#define CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK 1
+#endif // CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK
+
+namespace chip {
+namespace app {
+
+#if CHIP_DETAIL_LOGGING
+
+namespace {
+uint32_t gPrettyPrintingDepthLevel = 0;
```

Comment:
Should we add some notes regarding threading? I believe this assumes single threading.

Reply:
Done, thanks